visualization/app/codeCharta/util/deltaGenerator.ts

```diff
@@ -100,23 +104,29 @@ export class DeltaGenerator {
 		}
 	}
 
-	private static getDeltaAttributeList(referenceAttribute: KeyValuePair, comparisonAttribute: KeyValuePair) {
-		const deltaAttribute: KeyValuePair = {}
+	private static compareAttributeValues(reference: KeyValuePair, comparison: KeyValuePair) {
+		const deltaList: KeyValuePair = {}
+		let differenceExists = false
 
-		// TODO: All entries should have the combined attributes and deltas set,
-		// even if they do not exist on one side. Calculate these attributes up
-		// front. This operation is otherwise costly.
-		for (const key of Object.keys(comparisonAttribute)) {
-			deltaAttribute[key] = referenceAttribute[key] ? comparisonAttribute[key] - referenceAttribute[key] : comparisonAttribute[key]
-		}
+		const attributeKeys = [...Object.keys(reference), ...Object.keys(comparison)]
+
+		for (const key of attributeKeys) {
+			const referenceAttribute = reference[key]
+			const compAttribute = comparison[key]
 
-		for (const key of Object.keys(referenceAttribute)) {
-			if (!comparisonAttribute[key]) {
-				deltaAttribute[key] = -referenceAttribute[key]
+			// NOTE: One of these attributes will often be undefined
```

Comment:
Undefined should just be handled as "zero". That is in fact what you also do below, we should however already compare it with zeros. We do not want to count a undefined and zero entry as being changed as it would not be a noticable difference for the user. With the above, that's fine.

```suggestion
```

Reply:
Resolved with commit [5a13982](https://github.com/MaibornWolff/codecharta/pull/2890/commits/5a1398229e3209888d6ea4722b5334b6366d84ed)